test/unit/sync_mediator_test.rb

```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class SyncMediatorTest < KubernetesDeploy::TestCase
+  def setup
+    super
+    @fake_pod = FakePod.new('pod1')
+    @fake_deployment = FakeDeployment.new('deploy1')
+    @fake_cm = FakeConfigMap.new('cm1')
+    @fake_cm2 = FakeConfigMap.new('cm2')
+    @params = ['-a', '--output=json']
+  end
+
+  def test_get_instance_retrieves_resources_when_cache_is_empty
+    stub_kubectl_response('get', 'FakeConfigMap', @fake_cm.name, *@params, resp: @fake_cm.kubectl_response)
+    r = mediator.get_instance('FakeConfigMap', @fake_cm.name)
+    assert_equal @fake_cm.name, r.dig('metadata', 'name')
+
+    stub_kubectl_response('get', 'FakeConfigMap', 'does-not-exist', *@params, resp: {})
+    missing = mediator.get_instance('FakeConfigMap', 'does-not-exist')
+    assert_equal({}, missing)
+  end
+
+  def test_get_instance_uses_cache_when_available
+    resp = { "items" => [@fake_cm.kubectl_response, @fake_cm2.kubectl_response] }
+    stub_kubectl_response('get', 'FakeConfigMap', *@params, resp: resp)
+    mediator.get_all('FakeConfigMap')
+
+    # Only configmap is cached, so we should still make a request and get a result for a Deployment
+    stub_kubectl_response('get', 'FakeDeployment', @fake_deployment.name, *@params,
+      resp: @fake_deployment.kubectl_response)
+    uncached = mediator.get_instance('FakeDeployment', @fake_deployment.name)
+    assert_equal @fake_deployment.name, uncached.dig('metadata', 'name')
+
+    # Configmap is cached, so no new requests should happen here
+    mediator.kubectl.expects(:run).never
+    r = mediator.get_instance('FakeConfigMap', @fake_cm.name)
+    assert_equal @fake_cm.name, r.dig('metadata', 'name')
+    missing = mediator.get_instance('FakeConfigMap', 'does-not-exist')
+    assert_equal({}, missing)
+  end
+
+  def test_get_instance_does_not_populate_the_cache
```

Comment:
This test doesn't look finished, it also looks like its covered by the first test